### PR TITLE
Update

### PR DESCRIPTION
**Pull request by kasperg**
This PR introduces a bunch of different changes I made when digging into things. Sorry about providing them in one big PR - at least commits should be reasonably atomic. I suggest going through them one by one.

Key elements are:

- Implementation of business logic in controller
- Updates according to specification
- Updates in regard to how the package integrates with Laravel

For some reason test coverage does not pick up a single line. I cannot tell why 😖 .

### CODE REVIEW

File: tests/StatisticsControllerTest.php [hunk 1]
```diff
@@ -3,15 +3,86 @@
...
+            ->with('timestamp', '>=', $now)
+            ->once()
+            ->andReturnSelf()
+            ->shouldReceive('get')
+            ->andReturn(new Collection());
+
+        // FIXME: For some reason the test reports that the where() method
+        // does not exist.
+        // $response = $controller->get($request->reveal());
+        // $this->assertInstanceOf(Collection::class, $response);
```
Comment (kasperg) [inline tests/StatisticsControllerTest.php]: I would appreciate your help here @xendk

File: composer.json [hunk 2]
```diff
@@ -25,9 +25,19 @@
...
+        "orchestra/testbench": "^4.2",
         "phpstan/extension-installer": "^1.0",
         "phpstan/phpstan": "^0.11.16",
+        "phpstan/phpstan-mockery": "^0.11.3",
         "phpunit/phpunit": "^8.3",
         "squizlabs/php_codesniffer": "^3.5"
+    },
+    "extra": {
+        "laravel": {
+            "providers": [
```
Comment (xendk) [inline composer.json:38]: Oooh...

File: src/StatisticsController.php [hunk 3]
```diff
@@ -2,15 +2,49 @@
 
 namespace DDB\Stats;
 
+use Carbon\Carbon;
+use Illuminate\Database\DatabaseManager;
 use Illuminate\Http\Request;
+use Illuminate\Support\Facades\DB;
```
Comment (xendk) [inline src/StatisticsController.php]: This isn't needed anymore?

File: phpunit.xml [hunk 4]
```diff
@@ -9,6 +9,9 @@
             <directory suffix=".php">./src</directory>
         </whitelist>
     </filter>
+    <php>
+        <env name="DB_CONNECTION" value="testing"/>
```
Comment (xendk) [inline phpunit.xml:13]: Where's this connection defined? Looks handy.

File: routes/web.php [hunk 5]
```diff
@@ -12,4 +12,4 @@
 */
 
 /* @var \Laravel\Lumen\Routing\Router $router */
-$router->get('/statistics', '\\DDB\\Stats\\StatisticsController@get');
+$router->patch('/statistics', '\\DDB\\Stats\\StatisticsController@patch');
```
Comment (xendk) [inline routes/web.php:15]: Why PATCH? It's not modifying anything.

**Review by xendk — APPROVED** [review]

File: routes/web.php (hunk 5 above)
Comment (kasperg) [reply routes/web.php:15]: From the spec:

> By specifying a value an external system also signals that all preceeding events already have been claimed and can be deleted by the system.

File: phpunit.xml (hunk 4 above)
Comment (kasperg) [reply phpunit.xml:13]: I am not use to be honest. [The `testing` connection is mentioned in the techbench documentation](https://github.com/orchestral/testbench#overriding-setup-method).

**Pull request merged**